#! /bin/sh -e
## 08_privacy.dpatch by Eduard Bloch <blade@debian.org>
##
## DP: Remove cmdline from the isofs header

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
       -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
	*)
		echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
		exit 1;;
esac

exit 0
				    
@DPATCH@
diff -urNad /tmp/cdrtools-2.01/mkisofs/write.c cdrtools-2.01/mkisofs/write.c
--- /tmp/cdrtools-2.01/mkisofs/write.c	Wed May  7 17:17:45 2003
+++ cdrtools-2.01/mkisofs/write.c	Wed May  7 17:33:10 2003
@@ -2065,7 +2065,11 @@
 	}
 
 	cp[SECTOR_SIZE - 1] = '\0';
-	xfwrite(vers, SECTOR_SIZE, 1, outfile, 0, TRUE);
+ 	/* Per default: keep privacy. Blackout the version and arguments. */
+	if(getenv("ISODEBUG"))
+		xfwrite(vers, SECTOR_SIZE, 1, outfile, 0, TRUE);
+	else
+		xfwrite(calloc(SECTOR_SIZE, 1), SECTOR_SIZE, 1, outfile, 0, TRUE);
 	last_extent_written += 1;
 	return (0);
 }
